lib/src/metta/runner/stdlib.rs

```diff
@@ -707,6 +707,33 @@ impl Grounded for PrintlnOp {
     }
 }
 
+#[derive(Clone, PartialEq, Debug)]
+pub struct LogOp {}
+
+impl Display for LogOp {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "log!")
+    }
+}
+
+impl Grounded for LogOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_UNDEFINED, ATOM_TYPE_UNDEFINED, ATOM_TYPE_UNDEFINED])
+    }
+
+    fn execute(&self, args: &mut Vec<Atom>) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("log! expects two atoms as arguments");
+        let msg = args.get(0).ok_or_else(arg_error)?;
+        let val = args.get(1).ok_or_else(arg_error)?.clone();
```

Comment:
@vsbogd, I tried using `pop()` but I'm getting the error message "cannot borrow `*args` as mutable because it is also borrowed as immutable".

Reply:
Ah probably it is because you mixed `get` for the `msg` and `pop` for the `val`. First call to `get` returns the reference to the item of the `args` and thus borrows `args` for reading. Second `pop` requires mutable reference to `args` because it changes the collection state and it cannot get it because `args` is borrowed on the previous line.

You can use `pop` for getting both values in order to solve this.